include/nbl/builtin/hlsl/type_traits.hlsl

```diff
@@ -4,13 +4,425 @@
+  template<class T> struct is_scalar; (DONE)
+  template<class T> struct is_compound; (DONE)
+  template<class T> struct is_member_pointer; (TODO)
+ 
+  // type properties
+  template<class T> struct is_const; (DONE)
+  template<class T> struct is_volatile; (DONE)
+  template<class T> struct is_trivial; (EVERYTHING IS)
+  template<class T> struct is_trivially_copyable; (EVERYTHING IS)
+  template<class T> struct is_standard_layout; (NOT-APPLICABLE)
+  template<class T> struct is_empty; (DONE? sizeof(T) == 0)
+  template<class T> struct is_polymorphic; (NOTHING IS)
+  template<class T> struct is_abstract; (NOTHING IS)
+  template<class T> struct is_final; (NOTHING IS)
+  template<class T> struct is_aggregate; (DONE)
+ 
+  template<class T> struct is_signed; (DONE)
+  template<class T> struct is_unsigned; (DONE)
+  template<class T> struct is_bounded_array;
+  template<class T> struct is_unbounded_array;
+  template<class T> struct is_scoped_enum; (NOT-APPLICABLE)
+
+  // type property queries
+  template<class T> struct alignment_of; (TODO)
+  template<class T> struct rank; (TODO)
+  template<class T, unsigned I = 0> struct extent; (TODO)
+ 
+  // type relations
+  template<class T, class U> struct is_same; (DONE)
+  template<class Base, class Derived> struct is_base_of; (TODO)
+  template<class From, class To> struct is_convertible; (TODO)
+  template<class From, class To> struct is_nothrow_convertible; (NOT-APPLICABLE)
+  template<class T, class U> struct is_layout_compatible; (TODO)
+  template<class Base, class Derived> struct is_pointer_interconvertible_base_of; (NOT-APPLICABLE)
+ 
+  NO FOLD EXPRESSIONS IN HLSL!
+    template<class Fn, class... ArgTypes> struct is_invocable;
+    template<class R, class Fn, class... ArgTypes> struct is_invocable_r;
+    template<class Fn, class... ArgTypes> struct is_nothrow_invocable;
+    template<class R, class Fn, class... ArgTypes> struct is_nothrow_invocable_r;
+ 
+  // const-volatile modifications
+  template<class T> struct remove_const; (TODO)
+  template<class T> struct remove_volatile; (TODO)
+  template<class T> struct remove_cv; (TODO)
+  template<class T> struct add_const; (TODO)
+  template<class T> struct add_volatile; (TODO)
+  template<class T> struct add_cv; (TODO)
+
+  // reference modifications
+  template<class T> struct remove_reference; (TODO)
+  template<class T> struct add_lvalue_reference; (TODO)
+  template<class T> struct add_rvalue_reference; (TODO)
+
+  // sign modifications
+  template<class T> struct make_signed; (TODO)
+  template<class T> struct make_unsigned; (TODO)
+ 
+  // array modifications
+  template<class T> struct remove_extent; (TODO)
+  template<class T> struct remove_all_extents; (TODO)
+
+  // pointer modifications
+  template<class T> struct remove_pointer; (TODO)
+  template<class T> struct add_pointer; (TODO)
+
+  // other transformations
+  template<class T> struct type_identity; (TODO)
+  template<class T> struct remove_cvref; (TODO)
```

Comment:
I want these, but in a separate PR, some time later, either implement or open an issue

Reply:
although I guess I'd also need remove_reference, so nvm